clang/test/SemaSYCL/deferred-diagnostics-emit.cpp

```diff
@@ -0,0 +1,108 @@
+// RUN: %clang_cc1 -fcxx-exceptions -triple spir64 -fsycl-is-device -Wno-return-type -verify -fsyntax-only -std=c++17 %s
+
+/*
+    ensuring that the SYCL diagnostics that are typically deferred, correctly emit 
+*/
+
+struct S {
+  virtual void foo() {}
+};
+
+int calledFromKernel(int a){
+  // expected-error@+1 {{zero-length arrays are not permitted in C++}}
+  int MalArray[0];  
+
+  // expected-error@+1 {{__float128 is not supported on this target}}
+  __float128 malFloat = 40;  
+
+  // not sure if 'no virtual function' is a _deferred_ diagnostic, testing anyway 
+  S mal;
+  // expected-error@+1 {{SYCL kernel cannot call a virtual function}}
+  mal.foo();  
+
+  return a + 20;
+}
+
+inline namespace cl {
+namespace sycl {
+class queue {
+public:
+  template <typename T> void submit(T CGF) {}
+};
+
+template <int I> class id {};
+
+template <int I> class range {};
```

Comment:
AFAIK we already have mini fake SYCL runtime classes implementation in `test/SemaSYCL/Inputs/sycl.hpp` . I suggest reusing them.

Reply:
done